examples/evaluation/How_to_eval_abstractive_summarization.ipynb

```diff
@@ -0,0 +1,691 @@
+    "\"\"\"\n",
+    "\n",
+    "CONSISTENCY_SCORE_STEPS = \"\"\"\n",
+    "1. Read the article carefully and identify the main facts and details it presents.\n",
+    "2. Read the summary and compare it to the article. Check if the summary contains any factual errors that are not supported by the article.\n",
+    "3. Assign a score for consistency based on the Evaluation Criteria.\n",
+    "\"\"\"\n",
+    "\n",
+    "# Metric 4: Fluency\n",
+    "\n",
+    "FLUENCY_SCORE_CRITERIA = \"\"\"\n",
+    "Fluency(1-3): the quality of the summary in terms of grammar, spelling, punctuation, word choice, and sentence structure.\n",
+    "1: Poor. The summary has many errors that make it hard to understand or sound unnatural.\n",
+    "2: Fair. The summary has some errors that affect the clarity or smoothness of the text, but the main points are still comprehensible.\n",
+    "3: Good. The summary has few or no errors and is easy to read and follow.\n",
+    "\"\"\"\n",
+    "\n",
+    "FLUENCY_SCORE_STEPS = \"\"\"\n",
+    "Read the summary and evaluate its fluency based on the given criteria. Assign a fluency score from 1 to 3.\n",
+    "\"\"\"\n",
+    "\n",
+    "\n",
+    "def get_geval_score(\n",
+    "    criteria: str, steps: str, document: str, summary: str, metric_name: str\n",
+    "):\n",
+    "    prompt = EVALUATION_PROMPT_TEMPLATE.format(\n",
+    "        criteria=criteria,\n",
+    "        steps=steps,\n",
+    "        metric_name=metric_name,\n",
+    "        document=document,\n",
+    "        summary=summary,\n",
+    "    )\n",
+    "    response = openai.ChatCompletion.create(\n",
+    "        model=\"gpt-4\",\n",
+    "        messages=[{\"role\": \"user\", \"content\": prompt}],\n",
+    "        temperature=0,\n",
+    "        max_tokens=10,\n",
+    "        top_p=1,\n",
+    "        frequency_penalty=0,\n",
+    "        presence_penalty=0,\n",
+    "    )\n",
+    "    return response.choices[0].message.content\n",
+    "\n",
+    "\n",
+    "evaluation_metrics = {\n",
+    "    \"Relevance\": (RELEVANCY_SCORE_CRITERIA, RELEVANCY_SCORE_STEPS),\n",
+    "    \"Coherence\": (COHERENCE_SCORE_CRITERIA, COHERENCE_SCORE_STEPS),\n",
+    "    \"Consistency\": (CONSISTENCY_SCORE_CRITERIA, CONSISTENCY_SCORE_STEPS),\n",
+    "    \"Fluency\": (FLUENCY_SCORE_CRITERIA, FLUENCY_SCORE_STEPS),\n",
+    "}\n",
+    "\n",
+    "summaries = {\"Summary 1\": eval_summary_1, \"Summary 2\": eval_summary_2}\n",
+    "\n",
+    "data = {\"Evaluation Type\": [], \"Summary Type\": [], \"Score\": []}\n",
+    "\n",
+    "for eval_type, (criteria, steps) in evaluation_metrics.items():\n",
+    "    for summ_type, summary in summaries.items():\n",
+    "        data[\"Evaluation Type\"].append(eval_type)\n",
+    "        data[\"Summary Type\"].append(summ_type)\n",
+    "        result = get_geval_score(criteria, steps, excerpt, summary, eval_type)\n",
+    "        score_num = int(re.findall(r\"\\d+\\.?\\d*\", result)[0])\n",
```

Comment:
This doesn't seem very robust. Is there something better we could do here?

Reply:
refactored.